newIDE/app/src/ObjectEditor/Editors/SpriteEditor/SpritesList.js

```diff
@@ -177,30 +178,97 @@ type Props = {|
 |};
 
 const SpritesList = ({
+  spriteConfiguration,
   direction,
   project,
   resourcesLoader,
   resourceManagementProps,
-  onSpriteContextMenu,
-  selectedSprites,
-  onSelectSprite,
   onReplaceByDirection,
   onSpriteUpdated,
   onChangeName,
   objectName,
   animationName,
 }: Props) => {
   const [externalEditorOpened, setExternalEditorOpened] = React.useState(false);
+  // It's important to save the selected sprites in a ref, so that
+  // we can update the selection when a context menu is opened without relying on the state.
+  // Otherwise, the selection would be updated after the context menu is opened.
+  // Then, we need to ensure we trigger a force-update every time the selection changes.
+  const selectedSprites = React.useRef<{
+    [number]: boolean,
+  }>({});
+  const spriteContextMenu = React.useRef<?ContextMenuInterface>(null);
   const forceUpdate = useForceUpdate();
 
+  const updateIndexesAfterMoveUp = React.useCallback(

```

Comment:
I would name this `updateSelectionIndexesAfterMoveUp` (to emphasis that you're updating the selection, that's the job of the function)

Reply:
yep!